src/api/routes/stockRoute.js

```diff
@@ -10,6 +10,10 @@ export default class StockRoute {
         router.route('/stocks/users')
         .get((request, response) => 
         this.stockApiController.getStockWithUsers(request, response));
+
+        router.route('/stocks/analysis-data')
+        .post((request, response) =>
+        this.stockApiController.postStockAnalysisData(request.body, response));
```

Comment:
Is it possible for the request to be undefined?

Reply:
It can be, but it will just throw an error which I think should happen so that we ensure proper json body formatting.